pkg/output/http_writer.go

```diff
@@ -70,9 +71,16 @@ func (w *HTTPWriter) Write(p []byte) (n int, err error) {
 	bytesSent := len(p)
 	select {
 	case w.messages <- transport:
+		w.buffErrCount = 0
 	default:
 		slog.Error.Printf("Buffer full or closed, messages dropped")
 		metric.GetClient().Increment("http_writer.dropped_messages")
+
+		// Exit the pod if the error persists
+		if w.buffErrCount > 10 {
```

Comment:
@aamirtiwari can we have `const MaxBufferErrorCap = 10`

```suggestion
		if w.buffErrCount > MaxBufferErrorCap {
```